### ISSUE TITLE
class method formatted incorrectly due to missing context(?)

### ISSUE THREAD

**Issue by cmcaine** [nrwl/precise-commits#15]
```diff
diff --git a/src/keyseq.ts b/src/keyseq.ts
index 5caa81a..d5dccbc 100644
--- a/src/keyseq.ts
+++ b/src/keyseq.ts
@@ -35,12 +35,12 @@ export class MinimalKey {
 
     /** Does this key match a given MinimalKey extending object? */
     match(keyevent) {
-        // 'in' doesn't include prototypes, so it's safe for this object.
-        for (let attr in this) {
-            if (this[attr] !== keyevent[attr]) return false
-        }
-        return true
-    }
+                        // 'in' doesn't include prototypes, so it's safe for this object.
+                        for (let attr in this) {
+                            if (this[attr] !== keyevent[attr]) return false
+                        }
+                        return true
+                    }
 }
```

Prettier handles it correctly.

**Comment by JamesHenry** [comment]
@cmcaine Can I please check what you mean by prettier handles it correctly?

I am guessing you mean, when you run prettier on the whole file, and not just the range?

You can now reproduce ranges issues using the prettier playground.

E.g. Here is a range I am selecting from a separate, unrelated issue:
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgBtUAjfAMxwnXwB0RiT6BuWqNjqSKAZwIEN8AXnxkhAPnzceEQnAB0hCAHMAFCQCUrdjs6N5M9HBgALVFDVaQAGhARMMVNB7JQ-HFQDuABXcIXKPwAbhCoACY2ICQ4-GAA1sYAypix5srIMDgArnC25jxweN4xyuj8yOT8hAW2AFY8AB4AQjHxSfxGADLmcBVVNSApOAU4yFH8JACeSlCRmDjmMADq4abIABwADLbzEAVLMZhj83AjQb22OHAAjlmoV8X8peVIldW5IAXoqBnZHzxpOQARSyEHgfXethgExWYTWSAATFCYqhiBYAMLUMpjKDQC4gLIFAAqEwCbwGMQscES0LwyARSJAlOUcAAolAIkgAGybAC+vKAA

My guess is you will be able to reproduce the issue there if you apply the same range to your source that matches your git diff (i.e. what precise-commits is doing for you behind the scenes)

Please could you try and build that reproduction in the playground to confirm?

**Comment by cmcaine** [comment]
Yes, I am comparing with running the whole file through prettier.

This issue should maybe be on prettier rather than here. Depends what `prettier --range-*` is intended to do. Obviously in the precise-commits use this is wrong. Maybe if you just want to format some js in another document --range is fine, though what it's doing here is still pretty weird.

[Reproduction](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgGIQQ7AA6UJMA9AFTU4AiEcuMAFgJa4DWcAnjgFt0MMKxzocAc3YA3BDgCy7KOyGoA0nxxwAHvCgATZZJwQARgCs4YGAH4c1SmRwvBw0QAoevOHNgBKYmdXEMpKHAByZQicAyZMKAiYHGUwVABXAzgcAAcAJwgYQt4c5gAaHEwidhgI3Ex0ADNsxog8nDZOU0trGAA6YJCXVvaPVDhk4Rh25Q6OTEDSKCGVlMacD07MAG0pvIBdHABCAF4TnG9fBBhdmGn9wLyJ9LzlxvRUTDhBoYBfH5CTxgL2W03S32Wrn+UF+IDKIAgORg7GgmGQoHQeQKAHcAAqYhBolDoGQQdgGOEgMx5dBgHgwADKOVpxmQYLg8OUXzyMFxNMkQmQ70+HJAFkwOgAQjS6RMGegBHAADLKOBCj5feHMvLc5BU9BmXioaCU-LKGAAdXJbGQAA4AAxagpfC00nJ6-LMOB5OSUp4AR3S7CefPQAvQ6pF8K+AnYbLy4OjxnGAEV0oU1UhhZqQDADVaDDakAAmeHTdDsVDGADCEAEgpQUGgavh6S+ABUDUTs6KaVBJHAGXmecgAJy2+F9gcAUUMyAAzPb7b9fkA).
